cpp/src/arrow/type.cc

```diff
@@ -686,6 +686,57 @@ std::shared_ptr<Schema> schema(std::vector<std::shared_ptr<Field>>&& fields,
   return std::make_shared<Schema>(std::move(fields), metadata);
 }
 
+
+Status UnifySchemas(
+    const std::vector<std::shared_ptr<Schema>>& schemas,
+    std::shared_ptr<Schema>* out) {
+  if (schemas.empty()) {
```

Comment:
```suggestion
  DCHECK(!schemas.empty());
```

Reply:
did you mean only relying on a DCHECK?

I thought at some point we would want to make it a public function, then each language binding needs to do the check before calling it?